### ISSUE TITLE
Set up better training progress monitoring

### ISSUE THREAD

**Issue by EvgeniyaMartynova** [meszlili96/NaturalComputing#10]
At the moment I just save the images from the generator (in case of toy data set scatter plots) after some number of iterations. The only reason for it is that it is the simplest and the most straightforward way to implement it (for me).
In the examples, we worked with they used `torchvision.utils` and `tensorboardX`.  It would be good to use these tools in our project.

**Comment by NienkeWessel** [comment]
I worked on this for some hour and pushed code, but it is not quite finished yet. Apparently, TensorBoard does not work well with Matplotlib. It is not really possible to just save a matplotlib image to the TensorBoard. Two solutions I have found by googling are 
https://stackoverflow.com/questions/41356093/how-to-make-a-scatter-plots-using-tensorboard-tensorflow
and 
https://stackoverflow.com/questions/42329394/displaying-matplotlib-figure-in-tensorboard-instead-of-dumping-to-file
but neither is particularly nice or elegant. Not quite sure what the best way to go is.

Also, I've been having issues (of course) with the whole TensorBoard setup, but that is probably just my laptop again
